### ISSUE TITLE
Parsing Error When Specifying Varchar Lengths

### ISSUE THREAD

**Issue by taylorcmurphy** [reata/sqllineage#296]
Hey! Great project.

I'm running into an issue when trying to get column lineage against queries that specify varchar lengths – any columns that do so are excluded from the output.

```
INSERT INTO sample 
SELECT
a.col1::varchar(128) as does_not_show,
b.col2::varchar as will_show
from tabl1 as a
join tab2 as b
on a.i = b.i
;
```
output:
````
$ sqllineage -f example.sql -l column
<default>.sample.will_show <- <default>.tab2.col2

**Comment by reata** [comment]
This is PostgreSQL, correct?

Appending data type to column name is actually not officially supported yet. I didn't write any test case against it. I'm surprised that b.col2::varchar can be parsed. 

I need to take some time to review it.

**Comment by reata** [comment]
This issue is solved with the close of #364
